src/cpp/src/llm/pipeline_stateful.cpp

```diff
@@ -223,10 +215,28 @@ EncodedResults StatefulLLMPipeline::generate(
     ov::Tensor input_ids;
     ov::Tensor attention_mask;
     if (auto data = std::get_if<ov::Tensor>(&inputs)) {
+        if (m_is_npu) {
+            // Prefill model in NPU is reshaped to NPUW_LLM_MAX_PROMPT_LEN x NPUW_LLM_MAX_PROMPT_LEN
+            OPENVINO_ASSERT(data->get_size() <= m_max_prompt_len,
+                "Stateful LLM pipeline on NPU may only process prompts or hold chat history up to ",
+                m_max_prompt_len,
+                " tokens. ",
+                data->get_size(),
+                " is passed.\n Set the \"MAX_PROMPT_LEN\" config option to increase the limit.");
+        }
```

Comment:
Overall, I'd add a method here called `validate_input_length(size_t)` which'd do the required check in case of NPU. Then call it with whatsoever arguments.

Reply:
Thanks, will add!